sources/packages/backend/libs/utilities/src/logger/logger.service.ts

```diff
@@ -1,11 +1,49 @@
 import { Injectable, ConsoleLogger, Scope } from "@nestjs/common";
 import { LogLevels, ProcessSummary } from "./process-summary";
+import { StringBuilder } from "@sims/utilities/string-builder";
+import { parseJSONError } from "@sims/utilities/parse-json";
 
 /**
  * Common log across entire solution.
  */
 @Injectable({ scope: Scope.TRANSIENT })
 export class LoggerService extends ConsoleLogger {
+  /**
+   * Log a friendly error message.
+   * @param message friendly message.
+   */
+  error(message: string): void;
+  /**
+   * Converts the object into a string and log it.
+   * @param object object to be logged.
+   */
+  error(object: unknown): void;
+  /**
+   * Log the error message and the error details.
+   * @param message friendly message.
+   * @param error error captured.
+   * @param context optional log context.
+   */
+  error(message: string, error: unknown, context?: string): void;
+  /**
+   * Log the error message and the error details.
+   * @param message friendly message or object to be logged.
+   * @param error error captured.
+   * @param context optional log context.
+   */
+  error(message: unknown, error?: unknown, context?: string): void {
+    const errorBuilder = new StringBuilder();
+    if (typeof message === "string") {
+      errorBuilder.appendLine(message);
+    } else {
+      errorBuilder.appendLine(parseJSONError(message));
+    }
+    if (error) {
+      errorBuilder.appendLine(parseJSONError(error));
+    }
+    super.error(errorBuilder.toString(), context);
+  }
+
```

Comment:
For my understanding as well.

Reply:
Please see below the details.
Without the change using the below.
```ts
this.logger.error(`Error uploading file ${remoteFilePath}.`);
this.logger.error(error);
```
Results
![image](https://github.com/user-attachments/assets/2fa71eb2-cd45-4d0f-ac34-4e26be4442b7)

Using the below.
```ts
this.logger.error(`Error uploading file ${remoteFilePath}.`, error);
```
Results
![image](https://github.com/user-attachments/assets/6c0cba0e-c43f-4a25-b2ff-ae356f126f4d)

Using the PR code.
```ts
this.logger.error(`Error uploading file ${remoteFilePath}.`, error);
```
![image](https://github.com/user-attachments/assets/b794318f-b53d-485a-b598-d9d5d2af5676)

```ts
this.logger.error(`Error uploading file ${remoteFilePath}.`, error, "SOME CONTEXT");
```
![image](https://github.com/user-attachments/assets/ddd62d64-cd83-416d-8a9b-acd5ad562fc0)

Reply:
To be honest the `parseJSONError` was a quick implementation from the past to try to print errors and causes recursively.
It can be done in a better way but it is the method that we have right now.